### ISSUE TITLE
Default options for nasm under All Options

### ISSUE THREAD

**Issue by CodeVisio** [ShiftMediaProject/VSNASM#11]
Hi,
thanks for the tool. I really appreciate it.

However I'm facing an issue. By default nasm is invoked with some options as -Xvc and -f win32.
This is not always the case since not all projects containing an asm file to be processed by nasm are related to win32 and obj files or the other default options, -Xvc.

Is it possible to have by default only nasm invoked on an item? And under the properties of the project the possibility to insert or specify manually which options I'd like to have?

I've given a look at nasm.xml, .targets, and .props files and my intention was to modify them inorder to achieve the above goal but I don't know those formats (MSBuild?) and at the same time they seem to be interrelated.

Thanks

**Comment by Sibras** [comment]
Well -Xvc and -f win32 are required for the output to work with visual studio native windows tools and at the time this addon was written VS didnt have support for remote linux targets or the like which is what im assuming your trying to use it with?

**Comment by CodeVisio** [comment]
> linux targets or the like which is what im assuming your trying to use it with?
No

>Well -Xvc and -f win32 are required for the output to work with visual studio native windows tools
I didn't catch that.

Nasm can produce several output formats that are not necessarily for Windows nor even linux. Try nasm -hf. In the way the nasm.* files are configured instead they make the assumption that my ouput will always be win32 and this is not case.
Not making any assumptions, like instead is in nasm.props, about the nature of my projects and its output then you would give the user the options to select one (a drop list) that is more appropriate for his needs.
In other words, now the user is forced to use -f win32 and -Xvc and this is not always the case. With a drop list for -f flag instead you would give the user a choice about that. The same reason for the error report format, -X.

Anyway, did you build those files (nasm.*) by yourself? Or did you use a tool?

Thanks

**Comment by Sibras** [comment]
> In the way the nasm.* files are configured instead they make the assumption that my ouput will always be win32 and this is not case.

So if your output is not win32 what are you wanting to use? 

Any other output that is not win32 can not be understood by visual studio linker so will result in failed builds. So im just trying to understand your use case.

> Anyway, did you build those files (nasm.*) by yourself? Or did you use a tool?

I created them myself by hand (just good'ol notepad)

**Comment by CodeVisio** [comment]
> So if your output is not win32 what are you wanting to use?

-f bin

>Any other output that is not win32 can not be understood by visual studio linker so will result in
> failed builds. So im just trying to understand your use case.

Right. Indeed I'm not going to give it to the MS link. I mean, I can create an empty project with a bunch fo asm files and pass them to the nasm.exe producing a flat binary output thanks to your nasm.* files.
Giving default values options (-Xvc and -f win32) is not wrong, what is wrong is that I cannot change them on the properties dialog.

>I created them myself by hand (just good'ol notepad)

WOW, my hat off :)

**Comment by Sibras** [comment]
> -f bin

Ah OK I see, I havnt actually tested if that would work constantly though. VS can split the compilation of files into multiple separate nasm invocations for each source file which would result in them each getting output to a different output file. Without using a linker it would not be possible to get these files into a single output executable.

**Comment by CodeVisio** [comment]
I played a bit with xml targets and props, even recreating them from scratch without getting them working properly. I didn't find any good doc or even a tutorial. I also started from .rules from VS2008 and converting them to VS2017 without good result.
I've given a look at your xml targets and props files and they seem a copy of masm ones.

> Without using a linker it would not be possible to get these files into a single output executable.

If you assemble a single asm file into a flat binary's then you do not need any linker, link.exe or ld.

EDIT: typos

**Comment by Sibras** [comment]
> If you assemble a single asm file into a flat binary's then you do not need any linker, link.exe or ld.

Thats what I was referring to, it would only work for a single asm file and you wouldn't be able to have multiple files put in a single binary

**Comment by Sibras** [comment]
Since adding '-f bin' support only works in specific situations (i.e. only single asm file without using linker) then adding that to this customization will cause more issues than im prepared to deal with as the vast majority of the time users are not using this in that specific situation and would get consistent errors.

I suggest you either replace '-f win32' in your local copy or just leave it as is and use the VS linker (I dont see any harm in doing this)

**Comment by CodeVisio** [comment]
I agree. I found a workaround for just a single asm file.

Thanks.
